src/mcp_server_motherduck/server.py

```diff
@@ -38,11 +40,27 @@ def __init__(
         self.conn = self._initialize_connection()
         self.result_format = result_format
 
-    def _initialize_connection(self) -> duckdb.DuckDBPyConnection:
+    def _initialize_connection(self) -> Optional[duckdb.DuckDBPyConnection]:
         """Initialize connection to the MotherDuck or DuckDB database"""
 
         logger.info(f"🔌 Connecting to {self.db_type} database")
 
+        if self.db_type == "duckdb" and self._read_only:
+            # check that we can connect, issue a `select 1` and then close + return None
+            try:
+                conn = duckdb.connect(
+                    self.db_path,
+                    config={
+                        "custom_user_agent": f"mcp-server-motherduck/{SERVER_VERSION}"
+                    },
+                )
+                conn.execute("SELECT 1")
+                conn.close()
+                return None
+            except Exception as e:
+                logger.error(f"❌ Read-only check failed: {e}")
+                raise
+
```

Comment:
Should probably throw an error message here if `self._read_only=True` and point users to the 'saas_mode' flag for motherduck. Just to avoid that users falsely assume the read_only mode also applies to the md connection.